src/parameterizations/lateral/MOM_streaming_filter.F90

```diff
@@ -1,119 +1,200 @@
 
-!> This subroutine registers each of the fields to be filtered.
-subroutine Filt_register(a, om, grid, HI, CS)
-  real,                      intent(in)    :: a      !< Parameter that determines the bandwidth [nondim]
-  real,                      intent(in)    :: om     !< Target frequency of the filter [rad T-1 ~> rad s-1]
-  character(len=*),          intent(in)    :: grid   !< Horizontal grid location: h, u, or v
-  type(hor_index_type),      intent(in)    :: HI     !< Horizontal index type structure
-  type(Filter_CS),           intent(out)   :: CS     !< Control structure for the current field
+!> This subroutine registers the filter variables given the number of filters and the grid
+subroutine Filt_register(nf, key, grid, HI, CS, restart_CS)
+  integer,               intent(in)    :: nf           !< Number of filters to be used in the simulation
+  character(len=*),      intent(in)    :: key          !< Identifier of the variable to be filtered
+  character(len=*),      intent(in)    :: grid         !< Horizontal grid location: "h", "u", or "v"
+  type(hor_index_type),  intent(in)    :: HI           !< Horizontal index type structure
+  type(Filter_CS),       intent(out)   :: CS           !< Control structure of MOM_streaming_filter
+  type(MOM_restart_CS),  intent(inout) :: restart_CS   !< MOM restart control structure
 
   ! Local variables
-  integer :: isd, ied, jsd, jed, IsdB, IedB, JsdB, JedB
-
-  if (a  <= 0.0) call MOM_error(FATAL, "MOM_streaming_filter: bandwidth <= 0")
-  if (om <= 0.0) call MOM_error(FATAL, "MOM_streaming_filter: target frequency <= 0")
-
-  CS%a  = a
-  CS%om = om
+  type(axis_info) :: filter_axis(1)
+  real, dimension(:), allocatable :: n_filters         !< Labels of filters [nondim]
+  integer :: c
 
-  isd  = HI%isd  ; ied  = HI%ied  ; jsd  = HI%jsd  ; jed  = HI%jed
-  IsdB = HI%IsdB ; IedB = HI%IedB ; JsdB = HI%JsdB ; JedB = HI%JedB
+  CS%nf  = nf
+  CS%key = key
 
   select case (trim(grid))
     case ('h')
-      allocate(CS%s1(isd:ied,jsd:jed))   ; CS%s1(:,:) = 0.0
-      allocate(CS%u1(isd:ied,jsd:jed))   ; CS%u1(:,:) = 0.0
-      CS%is = isd  ; CS%ie = ied  ; CS%js = jsd  ; CS%je = jed
+      CS%is = HI%isd  ; CS%ie = HI%ied  ; CS%js = HI%jsd  ; CS%je = HI%jed
     case ('u')
-      allocate(CS%s1(IsdB:IedB,jsd:jed)) ; CS%s1(:,:) = 0.0
-      allocate(CS%u1(IsdB:IedB,jsd:jed)) ; CS%u1(:,:) = 0.0
-      CS%is = IsdB ; CS%ie = IedB ; CS%js = jsd  ; CS%je = jed
+      CS%is = HI%IsdB ; CS%ie = HI%IedB ; CS%js = HI%jsd  ; CS%je = HI%jed
     case ('v')
-      allocate(CS%s1(isd:ied,JsdB:JedB)) ; CS%s1(:,:) = 0.0
-      allocate(CS%u1(isd:ied,JsdB:JedB)) ; CS%u1(:,:) = 0.0
-      CS%is = isd  ; CS%ie = ied  ; CS%js = JsdB ; CS%je = JedB
+      CS%is = HI%isd  ; CS%ie = HI%ied  ; CS%js = HI%JsdB ; CS%je = HI%JedB
     case default
       call MOM_error(FATAL, "MOM_streaming_filter: horizontal grid not supported")
   end select
 
+  allocate(CS%s1(CS%is:CS%ie, CS%js:CS%je, nf)) ; CS%s1(:,:,:) = 0.0
```

Comment:
Rather than allocating and initializing arrays in two separate comments, the syntax we now prefer to use in MOM6 is to use the optional `source` argument to `allocate`, so the line 65 would become
`allocate(CS%s1(CS%is:CS%ie, CS%js:CS%je, nf), source=0.0)`
and similarly for line 66.